app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -87,6 +110,7 @@
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         username.setInputType(InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS | InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD);
+        editTextColor = username.getCurrentTextColor();

```

Comment:
recovering a color from a UI element seems like an odd pattern - are we basically using this to keep track of normal vs. red after bad login?